### PR TITLE
Support for additional request options applied to download request when bucket.upload() is called with URL

### PR DESCRIPTION
**Pull request by ImmaculatePine**
Implementation for #124 

- [x] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by ImmaculatePine** [comment]
I signed it! :)

File: src/bucket.js [hunk 1]
```diff
@@ -2099,6 +2099,9 @@ Bucket.prototype.setUserProject = function(userProject) {
  *     MD5 checksum for maximum reliability. CRC32c will provide better
  *     performance with less reliability. You may also choose to skip validation
  *     completely, however this is **not recommended**.
+ * @param {object} [options.requestOptions] When `pathString` is an URL additional
+ *     options for HTTP request could be provided here.
+ * @see [`request` options list](https://github.com/request/request#requestoptions-callback)
```
Comment (stephenplusplus) [inline src/bucket.js]: @callmehiphop are links still supported within a description? In other words:

... [options.requestOptions] When `pathString` is ...
... [options for HTTP request](http://)

Comment (ImmaculatePine) [reply src/bucket.js]: @stephenplusplus There are similar cases of links usage in this file, so I updated my PR.

**Comment by stephenplusplus** [comment]
Thanks @ImmaculatePine!

**Pull request merged**